temperature/crud.py

```diff
@@ -0,0 +1,65 @@
+import datetime
+
+from fastapi import HTTPException, Depends, status
+from fastapi.responses import JSONResponse
+from sqlalchemy.orm import Session
+from sqlalchemy.future import select
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.exc import IntegrityError
+from websockets.legacy import async_timeout
+
+from temperature import utils
+from weather import models
+from weather.crud import get_all_cities
+from dependencies import get_db
+
+
+async def get_temperature_list(db: AsyncSession):
+    result = await db.execute(
+        select(models.Temperature)
+    )
+    return result.scalars()
+
+
+async def get_temperatures_by_city_id(db: AsyncSession, city_id: int):
+    result = await db.execute(
+        select(models.Temperature).filter(models.Temperature.city_id == city_id)
+    )
+    temperature = result.scalars().first()
+
+    return temperature
+
+
+async def set_temperature(db: AsyncSession):
+    try:
+        await utils.check_service_connection()
+    except HTTPException as error:
+        raise error
+
+    data_list = await utils.get_temperature(db)
+    for city_name, city_data in data_list.items():
+        date_time_str = city_data["current"]["last_updated"]
+        date_time = datetime.datetime.strptime(date_time_str, "%Y-%m-%d %H:%M")
+        data = {
+            "city_id": city_data["city_id"],
+            "date_time": date_time,
+            "temperature": int(city_data["current"]["temp_c"]),
```

Comment:
The temperature value is being converted to an integer using `int(city_data['current']['temp_c'])`. Consider using `float()` instead to preserve decimal precision, as temperature values often include decimals.